editor/store/effects.js

```diff
@@ -94,6 +94,25 @@ export function removeProvisionalBlock( action, store ) {
 	}
 }
 
+/**
+ * Utilitiy used get an error object from a network response.
+ *
+ * @param {Object} response Response.
+ *
+ * @return {Promise} Error object promise.
+ */
+export const getErrorFromResponse = ( response ) => {
+	const getErrorFromBody = ( body ) => {
+		return body || {
+			code: 'unknown_error',
+			message: __( 'An unknown error occurred.' ),
+		};
+	};
+	return response.json()
+		.then( getErrorFromBody )
+		.catch( () => getErrorFromBody() );
```

Comment:
I wonder if it would be more useful to have this utility function within the apiFetch package itself?